### PR TITLE
docs: update state-management.mdx

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
State Management page is empty.

## What is the new behavior?
Updates page with all the details described in the original list.

## Other information

### CODE REVIEW

File: docs/core-concepts/state-management.mdx [hunk 1]
```diff
@@ -4,12 +4,121 @@ title: State Management
...
+// ---cut---
+import { createContext, createEffect, createSignal, useContext } from 'solid-start';
+import { Show, For } from 'solid-js';
+
+const ThemeContext = createContext(null);
+
+export function ThemeProvider() {
+  const INTITIAL_THEME = 'light';
+  const [theme, setTheme] = createSignal(INTIAL_THEME),
+    value = [
```
Comment (michaelessiet) [inline docs/core-concepts/state-management.mdx:40]: Add `const` to this, there's a syntax error here

File: docs/core-concepts/state-management.mdx [hunk 2]
```diff
@@ -4,12 +4,121 @@ title: State Management
...
+
+TBD...
+
+### Client-side stores
+
+TBD...normal Solid API stuff
+
+- createStore
+- Context
+- useSignal
```
Comment (michaelessiet) [inline docs/core-concepts/state-management.mdx:90]: createSignal

**Review by michaelessiet — CHANGES_REQUESTED** [review]

File: docs/core-concepts/state-management.mdx [hunk 3]
```diff
@@ -4,12 +4,121 @@ title: State Management
...
+      {
+        setTheme(theme) {
+          setTheme(theme);
+          localStorage?.setItem('theme', theme);
+        },
+      }
+    ];
+
+  createEffect(() => {
+    localStorage.getItem('theme') && setTheme(localStorage.getItem('theme'));
```
[the remark this answers is not in the document]
Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:51]: Oh man, great feedback. I normally prefer to write code that is readable from a junior up to a staff level and I really appreciate this accountability. 💯

File: docs/core-concepts/state-management.mdx (hunk 1 above)
Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:40]: uh oh, that means there is a [syntax error in the Solid docs too](https://www.solidjs.com/tutorial/stores_context?solved)! You are specifically talking about line 40, correct?

Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:40]: Done

File: docs/core-concepts/state-management.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:51]: Done

File: docs/core-concepts/state-management.mdx (hunk 2 above)
Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:90]: Done

File: docs/core-concepts/state-management.mdx [hunk 4]
```diff
@@ -4,12 +4,121 @@ title: State Management
...
+- Query parameters
+- Client-side stores
+
+### Local storage
+
+Local storage is a way to store data in the browser. This data is stored in the browser's local storage, and can be accessed by any page on the same domain. This data is persistent, and will remain even if the user closes their browser.
+
+When you need to access the `localStorage` API, you can either use a `createEffect` hook or a `createRouteAction` which gaurantees that the code will run in the browser.
+
+```tsx twoslash {19-21} filename="components/ThemePicker.tsx"
```
[the remark this answers is not in the document]
Comment (caseybaggz) [reply docs/core-concepts/state-management.mdx:29]: Done.

**Review by michaelessiet — APPROVED** [review]
Good work 👍

**Comment by ryansolid** [comment]
Thank you.

**Pull request merged**